jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2248,10 +2251,39 @@ void java_bytecode_convert_methodt::convert_athrow(
   codet &c,
   exprt::operandst &results) const
 {
-  side_effect_expr_throwt throw_expr;
-  throw_expr.add_source_location() = location;
-  throw_expr.copy_to_operands(op[0]);
-  c = code_expressiont(throw_expr);
+  if(
+    uncaught_exceptions_domaint::get_exception_type(op[0].type()) ==
+      "java::java.lang.AssertionError")
+  {
+    // we translate athrow into
+    // ASSERT false;
+    // ASSUME false:
+    code_assertt assert_code;
+    assert_code.assertion() = false_exprt();
+    source_locationt assert_location = location; // copy
```

Comment:
What is this copy good for?

Reply:
Because we use `location` as the basis for a modified source location.